# Check Title (`CheckClassName`)

A brief paragraph explaining why the check exists.

## Check Details

This check is aimed at eliminating ...

:-1: Examples of **incorrect** code for this check:

```liquid
```

:+1: Examples of **correct** code for this check:

```liquid
```

## Check Options

The default configuration for this check is the following:

```
CheckClassName:
  enabled: true
  some_option: 10
```

### `some_option`

The `some_option` option (Default: `10`) determines ...

## When Not To Use It

If you don't want to ..., then it's safe to disable this rule.

## Version

This check has been introduced in Theme Check X.X.X.

## Resources

- [Rule Source][codesource]
- [Documentation Source][docsource]

[codesource]: lib/theme_check/checks/check_class_name.rb
[docsource]: docs/checks/check_class_name.md
